pax-compiler/src/lib.rs

```diff
@@ -239,24 +82,55 @@ fn update_property_prefixes_in_place(manifest: &mut PaxManifest, host_crate_info
         updated_type_table.insert(t.0.replace("{PREFIX}", &host_crate_info.import_prefix), t.1.clone());
     });
     std::mem::swap(&mut manifest.type_table, &mut updated_type_table);
-
 }
 
+// The stable output directory for generated / copied files
+const PAX_DIR_PKG_PATH : &str = "pkg";
+
+fn clone_all_dependencies_to_tmp(pax_dir: &PathBuf, pax_version: &str, ctx: &RunContext) {
+
+    let dest_pkg_root = pax_dir.join(PAX_DIR_PKG_PATH);
+    for pkg in ALL_PKGS {
+
+        if ctx.is_libdev_mode {
+            //Copy all packages from monorepo root on every build.  this allows us to propagate changes
+            //to a libdev build without "sticky caches."
+            //
+            //Note that this may incur a penalty on libdev build times,
+            //since cargo will want to rebuild the whole workspace from scratch on every build.  If we want to optimize this,
+            //consider a "double buffered" approach, where we copy everything into a fresh new buffer (B), call it `.pax/pkg-tmp`, while leaving (A) `.pax/pkg`
+            //unchanged on disk.  Bytewise check each file found in B against a prospective match in A, and copy only if different.  (B) could also be stored on a virtual
+            //FS in memory, to reduce disk churn.
+            let pax_workspace_root = pax_dir.parent().unwrap().parent().unwrap();
+            let src = pax_workspace_root.join(pkg);
+            let dest = dest_pkg_root.join(pkg);
+            copy_dir_to(&src, &dest).expect(&format!("Failed to copy from {:?} to {:?}", src, dest));
+        } else {
+            let dest = dest_pkg_root.join(pkg);
+            if !dest.exists() {
+                let tarball_url = format!("https://crates.io/api/v1/crates/{}/{}/download", pkg, pax_version);
+                let resp = reqwest::blocking::get(&tarball_url)
+                    .expect(&format!("Failed to fetch tarball for {} at version {}", pkg, pax_version));
+
+                let tarball_bytes = resp.bytes().expect("Failed to read tarball bytes");
+                let mut tar = tar::Archive::new(&tarball_bytes[..]);
```

Comment:
Based on my local non-libdev run, it looks like this breaks due to crates.io sending back a gzipped tarball and tar expecting a uncompressed tar ball.  We'll likely need to add something like the following:
```
    let tar = GzDecoder::new(Cursor::new(bytes)); 
    let mut archive = Archive::new(tar);
```
We can punt this until another NUX coat of paint.